typegen/strats/inline.py

```diff
@@ -100,164 +114,240 @@ def _add_return_hint(self, fdef: ast.FunctionDef) -> None:
+            self.df[TraceData.FUNCNAME] == "",
+            self.df[TraceData.LINENO] == 0,
+            class_check,
+        ]
+
+        local_vars = self.df[functools.reduce(operator.and_, local_vars_mask)]
+        class_members = self.df[functools.reduce(operator.and_, class_member_mask)]
+
+        if len(targets) > 1:
+            hinted_targets: list[cst.BaseSmallStatement] = []
+            for ident, var in targets:
+                if isinstance(var, cst.Name):
+                    logger.debug(f"Searching for {ident} in local variables")
+                    hinted = local_vars[local_vars[TraceData.VARNAME] == ident]
+                else:
+                    logger.debug(f"Searching for {ident} in class attributes")
+                    hinted = class_members[class_members[TraceData.VARNAME] == ident]
+
+                assert (
+                    hinted.shape[0] <= 1
+                ), f"Found more than one type hint for {ident}"
+
+                if hinted.empty:
+                    logger.debug(f"No hint found for {ident}")
+                    continue
+
+                else:
+                    hint_ty = hinted[TraceData.VARTYPE].values[0]
+                    assert hint_ty is not None
+                    hint = hint_ty.__name__
+
+                    logger.debug(f"Found {hint} for {ident}")
+                    hinted_targets.append(
+                        cst.AnnAssign(
+                            target=var,
+                            annotation=cst.Annotation(cst.Name(value=hint)),
+                            value=None,
+                        )
+                    )
+
+            hinted_targets.append(original_node)
+            return cst.FlattenSentinel(hinted_targets)
+
+        ident, var = targets[0]
+        if isinstance(var, cst.Name):
+            hinted = local_vars[local_vars[TraceData.VARNAME] == ident]
+        else:
+            hinted = class_members[class_members[TraceData.VARNAME] == ident]
+
+        assert hinted.shape[0] <= 1, f"Found more than one type hint for '{ident}'"
+
+        if hinted.empty:
+            logger.debug(f"No hints found for '{ident}'")
+            return updated_node
+
+        hint_ty = hinted[TraceData.VARTYPE].values[0]
+        assert hint_ty is not None
+        hint = hint_ty.__name__
+
+        logger.debug(
+            f"Replacing Assign for '{ident}' with AnnAssign with hint '{hint}'"
+        )
+
+        # Replace simple assignment with annotated assignment
+        annass = cst.AnnAssign(
+            target=original_node.targets[0].target,
+            annotation=cst.Annotation(cst.Name(value=hint)),
+            value=original_node.value,
+        )
+
+        return annass
 
 
 class InlineGenerator(TypeHintGenerator):
     ident = "inline"
 
-    def _gen_hinted_ast(self, applicable: pd.DataFrame, nodes: ast.AST) -> ast.AST:
+    def _gen_hinted_ast(
+        self, applicable: pd.DataFrame, wrapper: cst.MetadataWrapper
```

Comment:
Change wrapper to original_ast.

Reply:
no, its literally an AST wrapped with Metadata

Reply:
`hintless_ast`